packages/mdc-segmented-button/segment/_segment-theme.scss

```diff
@@ -0,0 +1,114 @@
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+// Theme: Variables, functions, and mixins required to theme component segment
+
+@use "@material/feature-targeting/functions" as feature-targeting-functions;
+@use "@material/feature-targeting/mixins" as feature-targeting-mixins;
+@use "@material/theme/variables" as theme-variables;
+@use "@material/theme/custom-properties";
+
+$on-surface: theme-variables.prop-value(on-surface);
+$surface: theme-variables.prop-value(surface);
+$primary: theme-variables.prop-value(primary);
+
+$unselected-ink-color: rgba($on-surface, 0.6);
+$unselected-outline-color: rgba($on-surface, 0.12);
+$unselected-container-fill-color: rgba($surface, 1);
+$selected-ink-color: rgba($primary, 1);
+$selected-outline-color: $unselected-outline-color;
+$selected-container-fill-color: rgba($primary, 0.08);
+
+$height: 36px;
+$min-width: 48px;
+$horizontal-padding: 12px;
+$border-width: 1px;
+$border-radius: 4px;
+$icon-width: 24px;
+$icon-font-size: 18px;
+$label-padding: 6px;
+
+@mixin unselected-ink-color($color, $query: feature-targeting-functions.all()) {
+  .mdc-segmented-button__segment {
+    @include ink-color($color, $query);
+  }
+}
+
+@mixin unselected-container-color(
+  $color,
+  $query: feature-targeting-functions.all()
+) {
+  .mdc-segmented-button__segment {
+    @include container-fill-color($color, $query);
+  }
+}
+
+@mixin selected-ink-color($color, $query: feature-targeting-functions.all()) {
+  .mdc-segmented-button__segment.mdc-segmented-button__segment--selected {
+    @include ink-color($color, $query);
+  }
+}
+
+@mixin selected-container-color(
+  $color,
+  $query: feature-targeting-functions.all()
+) {
+  .mdc-segmented-button__segment.mdc-segmented-button__segment--selected {
+    @include container-fill-color($color, $query);
+  }
+}
+
+@mixin ink-color($color, $query: feature-targeting-functions.all()) {
+  $feat-color: feature-targeting-functions.create-target($query, color);
+
+  $property: custom-properties.create(--mdc-segmented-button-ink-color, $color);
```

Comment:
Move custom property create API to respective mixins (`selected-ink-color()` and `unselected-ink-color()`)  so that you can assign separate CSS custom property for each state - selected and unselected.

Make this mixin private (prefix with underscore).